code/game/machinery/cryopod.dm

```diff
@@ -0,0 +1,416 @@
+		if(general_record.fields["name"] == mob_occupant.real_name)
+			announce_rank = general_record.fields["rank"]
+			qdel(general_record)
+
+	control_computer?.frozen_crew += list(crew_member)
+
+	// Make an announcement and log the person entering storage.
+	if(GLOB.announcement_systems.len)
+		var/obj/machinery/announcement_system/announcer = pick(GLOB.announcement_systems)
+		announcer.announce("CRYOSTORAGE", mob_occupant.real_name, announce_rank, list())
+		visible_message("<span class='notice'>[src] hums and hisses as it moves [mob_occupant.real_name] into storage.</span>")
+
+
+	for(var/obj/item/item in mob_occupant.GetAllContents())
+		if(item.loc.loc && (item.loc.loc == loc || item.loc.loc == control_computer))
+			continue // means we already moved whatever this thing was in
+			// I'm a professional, okay
+		for(var/preserved in preserve_items)
+			if(istype(item, preserved))
+				if(control_computer && control_computer.allow_items)
+					control_computer.frozen_items += item
+					mob_occupant.transferItemToLoc(item, control_computer, TRUE)
+				else
+					mob_occupant.transferItemToLoc(item, loc, TRUE)
+
+	var/list/contents = mob_occupant.GetAllContents()
+	QDEL_LIST(contents)
+
+	// Ghost and delete the mob.
+	if(!mob_occupant.get_ghost(TRUE))
+		if(world.time < 15 MINUTES) // before the 15 minute mark
+			mob_occupant.ghostize(FALSE) // Players despawned too early may not re-enter the game
+		else
+			mob_occupant.ghostize(TRUE)
+	handle_objectives()
+	QDEL_NULL(occupant)
+	open_machine()
+	name = initial(name)
+
+/obj/machinery/cryopod/MouseDrop_T(mob/living/target, mob/user)
+	if(!istype(target) || !can_interact(user) || !target.Adjacent(user) || !ismob(target) || isanimal(target) || !istype(user.loc, /turf) || target.buckled)
+		return
+
+	if(occupant)
+		to_chat(user, "<span class='boldnotice'>[src] is already occupied!</span>")
+		return
+
+	if(target.stat == DEAD)
+		to_chat(user, "<span class='notice'>Dead people can not be put into cryo.</span>")
+		return
+
+	if(target.client && user != target)
+		if(iscyborg(target))
+			to_chat(user, "<span class='danger'>You can't put [target] into [src]. [target.p_theyre(capitalized = TRUE)] online.</span>")
+		else
+			to_chat(user, "<span class='danger'>You can't put [target] into [src]. [target.p_theyre(capitalized = TRUE)] conscious.</span>")
+		return
+	else if(target.client)
+		if(alert(target,"Would you like to enter cryosleep?",,"Yes","No") == "No")
+			return
+
+	var/generic_plsnoleave_message = " Please adminhelp before leaving the round, even if there are no administrators online!"
+
+	if(target == user && COOLDOWN_FINISHED(target.client, cryo_warned))
+		var/caught = FALSE
+		var/datum/antagonist/antag = target.mind.has_antag_datum(/datum/antagonist)
+		var/datum/job/target_job = SSjob.GetJob(target.mind.assigned_role)
+		if(target_job.req_admin_notify)
+			alert("You're a Head of Staff![generic_plsnoleave_message]")
```

Comment:
Your `req_admin_notify` check is correct, but this should be changed to something more generic like "You're an important role!"